central/main.go

```diff
@@ -651,19 +653,27 @@ func customRoutes() (customRoutes []routes.CustomRoute) {
 	})
 
 	scannerDefinitionsRoute := "/api/extensions/scannerdefinitions"
+	// Only grant compression to well-known content types. It should capture files
+	// worth of compression in definition's bundle. Ignore all other types (e.g.,
+	// `.zip` for the bundle itself).
+	definitionsFileGzipHandler, _ := gziphandler.GzipHandlerWithOpts(gziphandler.ContentTypes([]string{
+		mime.TypeByExtension(".json"),
+		mime.TypeByExtension(".yaml"),
```

Comment:
We should not leave it out. This piece is missing "text/plain" as well.

I am going to hard-code the content type because `mime.TypeByExtension()` actually relies on the distribution mime database, which is a dependency we don't want (that's why you got "" when you tested this in a sandbox).